### ISSUE TITLE
Invalid SPARQL Query can be executed via isql

### ISSUE THREAD

**Issue by jakubklimek** [openlink/virtuoso-opensource#312]
Commit: 9e0e2eb0f4a2516c41ec6722404584159b04c1ba
Database db01 provided earlier to @HughWilliams 

This query:

<pre><code> define output:format '_JAVA_'
  define input:same-as "yes"
 define input:default-graph-uri &lt;virt:DEFAULT&gt;
PREFIX  br:   &lt;http://purl.org/business-register#&gt;
PREFIX  mpsv: &lt;http://purl.org/mpsv/schools#&gt;
PREFIX  owl:  &lt;http://www.w3.org/2002/07/owl#&gt;
PREFIX  ros:  &lt;http://purl.org/register-of-schools#&gt;
PREFIX  gn:   &lt;http://www.geonames.org/ontology/&gt;
PREFIX  cm:   &lt;http://purl.org/czech-municipalities/geography#&gt;
PREFIX  gr:   &lt;http://purl.org/goodrelations/v1#&gt;
PREFIX  vcard: &lt;http://www.w3.org/2006/vcard/ns#&gt;
PREFIX  foaf: &lt;http://xmlns.com/foaf/0.1/&gt;
PREFIX  dc:   &lt;http://purl.org/dc/terms/&gt;

CONSTRUCT
  { ?s &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#type&gt; ros:School .
    ?s ros:facilityType ?schoolType .
    ?s ros:resortIdentificator ?resortIdentificator .
    ?s dc:description ?description .
    ?s mpsv:taughtLanguage ?taughtLanguage .
    ?s mpsv:contact ?contact .
    ?contact vcard:email ?email .
    ?contact vcard:tel ?tel .
    ?contact vcard:url ?web .
    ?s mpsv:contactPerson ?contactPerson .
    ?contactPerson foaf:name ?personName .
    ?contactPerson foaf:phone ?personPhone .
    ?s mpsv:priceForFood ?priceForFood .
    ?priceForFood gr:hasCurrencyValue ?foodPrice .
    ?s mpsv:priceForAccommodation ?priceForAccommodation .
    ?priceForAccommodation gr:hasCurrencyValue ?accommodationPrice .
    ?s ros:dateOfStart ?dateOfStart .
    ?s ros:dateOfRegistry ?dateOfRegistry .
    ?s ros:teachingLanguage ?schoolLanguage .
    ?s ros:capacity ?schoolCapacity .}
FROM &lt;http://ld.opendata.cz/resource/dataset/rejskol.msmt.cz&gt;
FROM &lt;http://ld.opendata.cz/resource/dataset/portal.mpsv.cz/skoly-evidovane-uradem-prace-cr&gt;
WHERE
  { ?s ros:facilityType ?schoolType .
    ?s ros:resortIdentificator "107518317"
    OPTIONAL
      { ?s dc:description ?description }
    OPTIONAL
      { ?s mpsv:taughtLanguage ?taughtLanguage }
    OPTIONAL
      { ?s mpsv:contact ?contact }
    OPTIONAL
      { ?contact vcard:email ?email }
    OPTIONAL
      { ?contact vcard:tel ?tel }
    OPTIONAL
      { ?contact vcard:url ?web }
    OPTIONAL
      { ?s mpsv:contactPerson ?contactPerson }
    OPTIONAL
      { ?contactPerson foaf:name ?personName }
    OPTIONAL
      { ?contactPerson foaf:phone ?personPhone }
    OPTIONAL
      { ?s mpsv:priceForFood ?priceForFood .
        ?priceForFood gr:hasCurrencyValue ?foodPrice
      }
    OPTIONAL
      { ?s mpsv:priceForAccommodation ?priceForAccommodation .
        ?priceForAccommodation gr:hasCurrencyValue ?accomodationPrice
      }
    ?s ros:dateOfStart ?dateOfStart .
    ?s ros:dateOfRegistry ?dateOfRegistry .
    ?s ros:teachingLanguage ?schoolLanguage
    OPTIONAL
      { ?s ros:capacity ?schoolCapacity }
  }</code></pre>

is invalid (try to run it through the SPARQL endpoint). However, when executed via isql, it runs indefinitely or for a long time - and I am not even sure what it does when the query is not valid.

**Comment by HughWilliams** [comment]
I have recreated with the DB provided and reported to development to fix ...

**Comment by HughWilliams** [comment]
@jakubklimek: Due to server crash we have lost the DB u provided for recreating this issue. Can u check to see if this still occurs with the latest develop/7 build please ?

**Comment by jakubklimek** [comment]
@HughWilliams  Yes, this issue still occurs in 492fc7cb48892760bd9db4c688caceea3419e09a

**Comment by HughWilliams** [comment]
@jakubklimek: Are you able to provide the database again, because as indicated in my previous post we lost the DB due to server machine failure which lost some of the test database ?
